media_service/tests/test_iiif.py

```diff
@@ -28,20 +28,40 @@ def setUp(self):
         self.factory = RequestFactory()
 
     def create_manifest(self, manifest_id, **kwargs):
-        request = self.factory.get('/')
-        return request, IIIFManifest(manifest_id, **kwargs)
+        return IIIFManifest(manifest_id, **kwargs)
 
     def get_images_list(self):
         images = [
             {'id': 1, 'is_link': False, 'url': 'http://localhost:8000/loris/foo.jpg', 'label': 'foo.jpg', 'description': '', 'metadata': []},
             {'id': 2, 'is_link': False, 'url': 'http://localhost:8000/loris/bar.jpg', 'label': 'bar.jpg', 'description': '', 'metadata': []},
             {'id': 3, 'is_link': True, 'url': 'http://my.link/image.jpg', 'label': 'image.jpg', 'description': '', 'metadata': []},
+            {
+                'id': 4,
+                'is_link': False,
+                'url': 'http://localhost:8000/loris/foo_bar_foo.png',
+                'label': 'foo_bar_foo.png',
+                'description': 'Foo Bar with Foo Again',
+                'metadata': [
+                    {"label": "Date", "value": "Once upon a time..."},
+                    {"label": "Location", "value": "In a galaxy far far away..."},
+                ]
+            },
         ]
         return images
+    
+    def duplicate_image(self, images, image_id, num_duplicates=1):
+        '''
+        Given a list of images, find an image by ID and duplicate it.
+        '''
+        image = next(iter([i for i in images if i['id'] == image_id]))
```

Comment:
Actually I think your way is simpler and more idiomatic. I just wanted a one-liner and for some reason I wasn't thinking of the list comprehension as an indexable list, but rather as an iterable, so that's why I reached for `next()` to return the first item. I don't think there's an advantage to mine for large lists either, since it's not a generator.